ui/src/text_annotator.tsx

```diff
@@ -227,22 +265,53 @@ export
             <span style={removeIconStyle as React.CSSProperties} className={css.iconUnderlay}></span>
           </mark>
         )
-      }
+      },
+      handleOnMouseUp = (ev: React.MouseEvent<HTMLSpanElement>) => {
+        if (smart_selection) annotate()
+        else if (timeoutRef.current) annotate(ev.currentTarget) // double-click
+        else if (new Date().getTime() - mouseDownTimeRef.current > 250) annotate() // dragging (long click)
+        else timeoutRef.current = setTimeout(() => annotate(), 250) // click
```

Comment:
Maybe this could be a better detection for double click? https://stackoverflow.com/questions/5497073/how-to-differentiate-single-click-event-and-double-click-event/53939059#53939059